framework/helpers/BaseHtml.php

```diff
@@ -1719,7 +1719,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         );
         $options['encode'] = ArrayHelper::getValue($options, 'encode', $encode);
 
-        if (null !== ($prompt = ArrayHelper::remove($tagOptions, 'prompt'))) {
+        $prompt = ArrayHelper::remove($tagOptions, 'prompt');
+        if (null !== $prompt) {
```

Comment:
you are validating $prompt against something, not the other way around.
It should be:
```php
if($prompt !== null){    //which reads like: if 'prompt' is not null.....

}

``